server.js

```diff
@@ -0,0 +1,218 @@
+      let decodeChunk = querystring.parse(body);
+      // Used for new element pages
+      addOrUpdateElem(decodeChunk, req, res)
+      // Update index.html
+      parseNewHTML(decodeChunk, req, res);
+      res.writeHead(200, {
+        "Content-Type": 'application/json',
+        "Success": true,
+        "Date": date
+      })
+      res.end();
+    }
+    /*** PUT Method ***/
+    if (req.method === "PUT") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      fs.readdir("./public/", (err, files) => {
+        if (err) {
+          console.log(err)
+        }
+        if (files.includes(req.url.slice(1)) === true) {
+          addOrUpdateElem(decodeChunk, req, res)
+        } else {
+          res.writeHead(500, {
+            "Content-Type": `application/json, charset=utf-8`,
+            "error": `resource ${req.url} does not exist`,
+            "Date": date
+          });
+          res.end();
+        }
+      });
+    }
+    /*** DELETE Method ***/
+    if (req.method === "DELETE") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      fs.readdir("./public/", (err, files) => {
+        if (err) {
+          return console.log(err);
+        }
+        if (files.includes(req.url.slice(1)) === true) {
+          fs.unlink(`./public${req.url}`, (err) => {
+            parseNewHTML(decodeChunk, req, res);
+          });
+        } else {
+          res.writeHead(500, {
+            "Content-Type": `application/json, charset=utf-8`,
+            "error": `resource ${req.url} does not exist`,
+            "Date": date
+          });
+          res.end();
+        }
+      });
+      res.end();
+    }
+  });
+});
+
+server.listen(PORT, () => {
+  console.log(`Server started on PORT: ${PORT}`);
+});
+
+/*************** FUNCTIONS **************/
+
+// Add and/or Update element.html
+function addOrUpdateElem(chunk, req, res) {
+  // Used for new element pages
+  let htmlTemplate = `<!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8" />
+            <title>The Elements - ${chunk["elementName"]}</title>
+            <link rel="stylesheet" href="/css/styles.css" />
+          </head>
+          <body>
+            <h1>${chunk["elementName"]}</h1>
+            <h2>${chunk["elementSymbol"]}</h2>
+            <h3>Atomic number ${chunk["elementAtomicNumber"]}</h3>
+            <p>
+              ${chunk["elementDescription"]}
+            </p>
+            <p><a href="/">back</a></p>
+          </body>
+        </html>`;
+  // Create new element page
+  fs.writeFile(
+    `./public/${chunk["elementName"]}.html`,
+    htmlTemplate,
+    err => {
+      if (err) {
+        return console.log("Cannot write file " + err);
+      }
+    }
+  );
+  res.writeHead(200, {
+    "Content-Type": `text/html, charset=utf-8`,
+    "Content-Length": htmlTemplate.length,
+    "Date": date
+  });
+  res.end();
```

Comment:
Instead of ending here, you should probably either pass in the parse function if it needs to be done, or another function to handle the res.writeHead/end